<!-- https://github.com/ros-controls/ros_controllers/tree/kinetic-devel/diff_drive_controller/test -->
<launch>
    <arg name="model" default="fellowme"/>
    <arg name="description_package" default="$(eval find(arg('model') + '_description'))" />
    <arg name="urdf_file" default="$(find xacro)/xacro '$(arg description_package)/urdf/$(arg model).urdf.xacro'" />

    <!-- Load the URDF model of fellowme into the ROS Parameter Server -->
    <param name="fellowme/robot_description" command="$(arg urdf_file)" />

    <!-- Run the fellowme_base -->
    <!-- This will create the fellowme hardware interface and instantiate a new controller manager with it -->
    <node name="fellowme_base" pkg="fellowme_base" type="fellowme_base" ns="fellowme" output="screen"/>

    <!-- Load controller config to the parameter server -->
    <rosparam command="load" file="$(find fellowme_control)/config/fellowme_control.yaml"/>

    <!-- Load base config to the parameter server -->
    <rosparam command="load" 
              file="$(find fellowme_base)/config/base.yaml"/>

    <!-- Load the controllers -->
    <!-- This will use the controller manager that got started inside the fellowme_base node -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
        output="screen" ns="fellowme" args="mobile_base_controller"/>
</launch>